state/indexer/block/kv/util.go

```diff
@@ -108,25 +107,42 @@ func parseHeightFromEventKey(key []byte) (int64, error) {
 
 func parseEventSeqFromEventKey(key []byte) (int64, error) {
 	var (
-		compositeKey, typ, eventValue string
-		height                        int64
-		eventSeq                      int64
+		compositeKey, eventValue string
+		height                   int64
+		eventSeq                 int64
 	)
 
-	remaining, err := orderedcode.Parse(string(key), &compositeKey, &eventValue, &height, &typ)
+	remaining, err := orderedcode.Parse(string(key), &compositeKey, &eventValue, &height)
 	if err != nil {
-		return 0, fmt.Errorf("failed to parse event key: %w", err)
+		return 0, fmt.Errorf("failed to parse event sequence: %w", err)
 	}
 
-	// This is done to support previous versions that did not have event sequence in their key
+	// We either have an event sequence or a function type (potentially) followed by an event sequence.
+	// Potential scenarios:
+	// 1. Events indexed with v0.38.x and later, will only have an event sequence
+	// 2. Events indexed between v0.34.27 and v0.37.x will have a function type and an event sequence
+	// 3. Events indexed before v0.34.27 will only have a function type
+	// function_type = 'being_block_event' | 'end_block_event'
```

Comment:
I'd suggest putting the remaining == 0 case first and reducing the indentation level of the what's in the current if block.

```
if len(remaining) != 0 {
		return 0, fmt.Errorf("failed to parse event sequence, invalid event format")
	}
```